compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -69,21 +71,29 @@ pub trait InferCtxtEvalExt<'tcx> {
 }
 
 impl<'tcx> InferCtxtEvalExt<'tcx> for InferCtxt<'tcx> {
+    #[instrument(level = "debug", skip(self))]
     fn evaluate_root_goal(
         &self,
         goal: Goal<'tcx, ty::Predicate<'tcx>>,
     ) -> Result<(bool, Certainty), NoSolution> {
         let mut search_graph = search_graph::SearchGraph::new(self.tcx);
 
-        let result = EvalCtxt {
+        let mut ecx = EvalCtxt {
             search_graph: &mut search_graph,
             infcx: self,
             // Only relevant when canonicalizing the response.
             max_input_universe: ty::UniverseIndex::ROOT,
             var_values: CanonicalVarValues::dummy(),
-            in_projection_eq_hack: false,
+            nested_goals: NestedGoals::new(),
+        };
+        let result = ecx.evaluate_goal(IsNormalizesToHack::No, goal);
+
+        if let Ok((_, Certainty::Yes)) = result {
```

Comment:
even without `Certainty::Yes`, we don't do anything with the nested goals, so they have to be always empty... or we return them to the caller, but we must not simply drop them here

Reply:
I guess `evaluate_added_goals_and_make_canonical_response` could just empty `NestedGoals`

Reply:
`evaluate_root_goal` doesn't use `evaluate_added_goals_and_make_canonical_response`? :thinking:

Reply:
ah xd